### DISCUSSION TITLE
antsApplyTransforms doesn't give the same results as antsRegistrationSyn

### DISCUSSION THREAD

**Discussion by mrspsy** [ANTsX/ANTs#1624] [Q&A] [answered]
Hello, 

I'm trying to use ANTs to register and align 3D-whole brain MR spectroscopic imaging metabolic maps to T1. 
AntsRegistrationSyn does give me a pretty good result using this command : 

antsRegistrationSyN.sh -d 3 -m OrigRes_Cr+PCr_conc.nii.gz -f rA009_V5_T1_brain.nii.gz -t sr -o Ants_CrtoT1_SyNsr -n 6 

Then I want to apply the same transformation to other metabolic maps / SNR or CRLB maps corresponding to the metabolites using antsApplyTransforms. But it seems to not work, the results are not as well aligned as with antsRegistration. I wanted to see if it's due to the inherent quality of the other maps or if it was a problem with Ants. So I applied the same transformation to the initial map that I registered on antsRegistrationSyn to see if I get the same result, using this command : 
antsApplyTransforms -d 3 -i OrigRes_Cr+PCr_conc.nii.gz -r rA009_V5_T1_brain.nii.gz -o ApplyCrAffine+WarpToCr_DummyTest.nii.gz -n Linear -t Ants_CrtoT1_SyNsr0GenericAffine.mat -t Ants_CrtoT1_SyNsr1Warp.nii.gz -v

And with this I get the same problem. So, to summarize, beginning with the same map (OrigRes_Cr+PCr_conc), wether I use antsRegistrationSyn, or antsApplyTransforms with the Affine and warp created with antsRegistratinoSyn minutes ago, I don't get the same result. 
I tried to use the option "repro" (-y 1) but I get the same issue. 

Is it normal ? Do I miss a step ? 

Thank you for your answers ! 

Edgar

**Comment by cookpa** [comment, marked as answer]
Try switching the transform order: `antsApplyTransforms -d 3 -i OrigRes_Cr+PCr_conc.nii.gz -r rA009_V5_T1_brain.nii.gz -o ApplyCrAffine+WarpToCr_DummyTest.nii.gz -n Linear -t Ants_CrtoT1_SyNsr1Warp.nii.gz -t Ants_CrtoT1_SyNsr0GenericAffine.mat -v`

https://github.com/ANTsX/ANTs/wiki/Forward-and-inverse-warps-for-warping-images,-pointsets-and-Jacobians

**Comment by mrspsy** [comment]
Thanks a lot for your quick answer, it works perfectly ! I assumed that it would be the other way around and didn't question this point... Thanks again :)
